lib/pbench/cli/server/user_update.py

```diff
@@ -0,0 +1,107 @@
+import click
+from pbench import BadConfig
+from pbench.server import PbenchServerConfig
+from pbench.cli.server import pass_cli_context
+from pbench.cli.server.options import common_options
+from pbench.common.logger import get_pbench_logger
+from pbench.server.database.models.users import User
+from pbench.server.database.database import Database
+
+_NAME_ = "pbench-user-update"
+
+
+class UserUpdate:
+    def __init__(self, context):
+        self.context = context
+
+    def execute(self):
+        config = PbenchServerConfig(self.context.config)
+
+        logger = get_pbench_logger(_NAME_, config)
+
+        # We're going to need the Postgres DB to track dataset state, so setup
+        # DB access.
+        Database.init_db(config, logger)
+
+        # Check if the user has the right privileges
+        if user is None or not user.is_admin():
+            raise RuntimeError(f"Not authorized to update user {self.context.username}")
+
+        dict_to_update = {}
+        if self.context.username:
+            dict_to_update["username"] = self.context.username
+
+        if self.context.first_name:
+            dict_to_update["first_name"] = self.context.first_name
+
+        if self.context.last_name:
+            dict_to_update["last_name"] = self.context.last_name
+
+        if self.context.email:
+            dict_to_update["email"] = self.context.email
+
+        if self.context.role:
+            dict_to_update["role"] = self.context.role
+
+        # If the authorization passed update the user from the given username
+        User.update(dict_to_update)
+
+        click.echo(f"User {self.context.user_to_update} updated")
+
+
+@click.command()
+@common_options
+@click.option(
+    "--user",
+    required=True,
+    prompt=True,
+    help="pbench server account username (will prompt if unspecified)",
+)
+@click.option(
+    "--user-to-update",
+    prompt=True,
+    required=True,
+    help="Specify username of a pbench server account user to update (will prompt if unspecified)",
+)
+@click.option(
+    "--username", required=False, help="Specify the new username",
+)
+@click.option(
+    "--email", required=False, help="Specify the new email",
+)
```

Comment:
As I suggested for user-delete, I would recommend specifying the to-be-updated user (whether that's via a user-ID, a username, or an email address) as the command `parameter`, and use `option`'s to indicate the values that we want to change.

(We could consider being flexible about how we interpret the parameter -- we could do a sequence of look-ups in different columns if we don't find it on the first try -- that might provide nice flexibility for the user.)